src/main/scala/org/apache/spark/sql/pulsar/PulsarMetadataReader.scala

```diff
@@ -392,7 +393,7 @@ private[pulsar] case class PulsarMetadataReader(
         } else {
           assert (time > 0, s"time less than 0: $time")
           if (client == null) {
-            client = PulsarClient.builder().serviceUrl(serviceUrl).build()
+            client = createPulsarClient(serviceUrl, clientConf)
```

Comment:
https://github.com/streamnative/pulsar-spark/blob/87b1cb8a7c62cf5100a91a5ba568ad17997483c2/src/main/scala/org/apache/spark/sql/pulsar/PulsarMetadataReader.scala#L462 This line should also be modified to use `createPulsarClient`

Reply:
This had been removed, since we will be using a `CachedPulsarClient` for this as well.